python/paddle/fluid/incubate/fleet/base/role_maker.py

```diff
@@ -281,25 +293,30 @@ def _server_num(self):
         return the current number of server
         """
         if self._check_role_generation():
-            if self.is_server():
-                return self._get_size() / 2
-        return 0
+            return self._get_size() / 2
+        else:
+            self.generate_role()
+            return self._get_size() / 2
 
     def worker_index(self):
         """
         return the index of worker
         """
         if self._check_role_generation():
             return self._rank / self._proc_per_node
```

Comment:
what is the difference between self._proc_per_node with `2`?

Reply:
fixed